### PR TITLE
Move coverage from travis to prow - change travis, update builder

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

This PR removes the coverage run from the travis build and prepares setting environment variables for running it as a prow job. See https://github.com/kubevirt/project-infra/pull/586 for the prow job that uses the changes applied here.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
I had to apply some pretty ugly workarounds to get the builder image building. https://github.com/kubevirt/kubevirt/pull/4011/commits/ae2527a37ed3ef5137087825c7c0a5171e4824f7 due to [some dependency](https://github.com/fvbommel/util/issues/6) not getting resolved.

Created https://github.com/kubevirt/kubevirt/issues/4037 to track ppc64le disabling wich was required due to packages missing when building the builder image.

**Release note**:

```release-note
ppc64le arch has been disabled for the moment, see https://github.com/kubevirt/kubevirt/issues/4037
```

### CODE REVIEW

File: automation/travisci-test.sh
Comment (xpivarc) [inline automation/travisci-test.sh:11]: do we want to run this?

Comment (dhiller) [reply automation/travisci-test.sh:11]: We should run this at least as long as the prow job `pull-kubevirt-unit-test` is not yet reporting to GitHub I think.

File: hack/builder/Dockerfile
Comment (xpivarc) [inline hack/builder/Dockerfile:29]: We need to keep this. https://travis-ci.org/github/kubevirt/kubevirt/builds/720719969#L428

**Review by xpivarc — CHANGES_REQUESTED** [review]
Hopefully one last functional change we need to do here. (:crossed_fingers: )

File: hack/builder/Dockerfile
Comment (dhiller) [reply hack/builder/Dockerfile:29]: That package was removed from fc31. Had to add a manual install to reenable it.

**Comment by dhiller** [comment]
Coverage reported from the tested prow job, everything seems to work now.

File: hack/builder/Dockerfile
Comment (rmohr) [inline hack/builder/Dockerfile:16]: The bazel version is also gone? Which bazel ersion do we get now? Please hardcode it to avoid surpises (which I had in the past).

**Review by rmohr — CHANGES_REQUESTED** [review]
One request, the rest looks fine. :+1:

**Comment by rmohr** [comment]
/hold
We need to make the coveralls job required before merging this.

**Comment by dhiller** [comment]
> /hold
> We need to make the coveralls job required before merging this.

I think you mean that we need to make it `always_run: true`, right? The `coverage/coveralls` check is already required, for which the other job is posting results, so no need to make that required. But at least we maybe should make it report to github also? WDYT?

**Comment by dhiller** [comment]
Triggered the coverage job here again: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4011/pull-kubevirt-goveralls/1298285142132920320

**Comment by rmohr** [comment]
> I think you mean that we need to make it `always_run: true`, right? The `coverage/coveralls` check is already required, for which the other job is posting results, so no need to make that required. But at least we maybe should make it report to github also? WDYT?

I think the job must be required and reporting too. Otherwise people may end up with not reported coveralls or failed coveralls and they can't just do a simple /retest.

**Comment by rmohr** [comment]
> I think the job must be required and reporting too. Otherwise people may end up with not reported coveralls or failed coveralls and they can't just do a simple /retest.

We may then mark with our maintenance tool on all PRs where coveralls succeded the new required job as passed to avoid confusion on existing PRs.

**Comment by dhiller** [comment]
> > I think you mean that we need to make it `always_run: true`, right? The `coverage/coveralls` check is already required, for which the other job is posting results, so no need to make that required. But at least we maybe should make it report to github also? WDYT?
> 
> I think the job must be required and reporting too. Otherwise people may end up with not reported coveralls or failed coveralls and they can't just do a simple /retest.

Done: https://github.com/kubevirt/project-infra/pull/586/files#diff-b6d3f192a82fea00b9a4e135a8dcbe4bR734

**Comment by rmohr** [comment]
/hold cancel
Merging manually with https://github.com/kubevirt/project-infra/pull/586 to cleanly migrate the job.

**Pull request merged**
